src/AlertsManagement/AlertsManagement/AlertCommands/GetAzureAlert.cs

```diff
@@ -0,0 +1,340 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Linq;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Rest.Azure;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.AlertsManagement.OutputModels;
+using Microsoft.Azure.Management.AlertsManagement.Models;
+
+namespace Microsoft.Azure.Commands.AlertsManagement
+{
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "Alert", DefaultParameterSetName = AlertsListByFilterParameterSet)]
+    [OutputType(typeof(PSAlert))]
+    public class GetAzureAlertCommand : AlertsManagementBaseCmdlet
+    {
+        #region Parameter Set Names
+
+        private const string AlertsListByFilterParameterSet = "AlertsListByFilter";
+        private const string AlertsListByTargetResourceIdFilterParameterSet = "AlertsListByTargetResourceIdFilter";
+        private const string AlertsListByTargetResourceGroupFilterParameterSet = "AlertsListByTargetResourceGroupFilter";
+        private const string AlertByIdParameterSet = "AlertById";
+
+        #endregion
+
+        #region Parameters declarations
+
+        /// <summary>
+        /// Alert Id
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   ParameterSetName = AlertByIdParameterSet,
+                   HelpMessage = "Unique Identifier of Alert / ResourceId of alert.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("ResourceId")]
+        public string AlertId { get; set; }
```

Comment:
Pre the PR review, this should also have a 'byName' parameter set that allows getting a particular alert by rgname + name (or rgname +id, as the case may be), unless resource group name is not necessary to cal up by ig (Guid)

Reply:
Alerts are proxy resources - they do not come under any particular resource group. Only Id is required for fetching alert.